web/client/reducers/geostory.js

```diff
@@ -317,7 +317,8 @@ export default (state = INITIAL_STATE, action) => {
         }
 
         if (isArray(oldElement) && isArray(newElement) && mode === "merge") {
-            newElement = [ ...oldElement, ...newElement ];
+            // check for options to filter unique options
+            newElement = (options?.uniqueByKey) ? uniqBy([ ...oldElement, ...newElement ], options.uniqueByKey) : [...oldElement, ...newElement ];
```

Comment:
please add also a test for the reducer, thanks

Reply:
Got it, working on it.

Reply:
I'm not seeing the test in the reducers but a new one in the plugin, please add a new unit test for this case in the reducer https://github.com/geosolutions-it/MapStore2/blob/master/web/client/reducers/__tests__/geostory-test.js